ttnn/cpp/ttnn/operations/ccl/erisc_datamover_builder.cpp

```diff
@@ -0,0 +1,414 @@
+    sender_channel_0_connection_semaphore_id(sender_channel_0_connection_semaphore_id),
+    sender_channel_1_connection_semaphore_id(sender_channel_1_connection_semaphore_id),
+    sender_channel_0_buffer_index_semaphore_id(sender_channel_0_buffer_index_semaphore_id),
+    sender_channel_1_buffer_index_semaphore_id(sender_channel_1_buffer_index_semaphore_id),
+
+    receiver_channel_local_buffer_index_addr(FabricEriscDatamoverConfig::receiver_channel_local_buffer_index_addr),
+
+    local_sender_channel_0_buffer_address(config.sender_0_channel_base_address),
+    local_sender_channel_0_connection_info_addr(
+        FabricEriscDatamoverConfig::sender_channel_0_worker_connection_info_address),
+    local_sender_channel_1_buffer_address(config.sender_1_channel_base_address),
+    local_sender_channel_1_connection_info_addr(
+        FabricEriscDatamoverConfig::sender_channel_1_worker_connection_info_address),
+    local_receiver_channel_buffer_address(config.receiver_channel_base_address),
+
+    termination_signal_ptr(FabricEriscDatamoverConfig::termination_signal_address) {}
+
+std::vector<uint32_t> FabricEriscDatamoverBuilder::emit_compile_time_args() const {
+    const bool is_handshake_master = this->my_chip_id < this->peer_chip_id;
+    TT_ASSERT(this->my_chip_id != this->peer_chip_id);
+    TT_ASSERT(
+        this->sender_0_num_buffers == this->sender_1_num_buffers);  //, "Implementation expects sender_0_num_buffers and
+                                                                    // sender_1_num_buffers to be the same for now");
+    log_trace(tt::LogTest, "Sender 0 num buffers: {}", this->sender_0_num_buffers);
+    log_trace(tt::LogTest, "Sender 0 channel address: {}", this->local_sender_channel_0_buffer_address);
+    log_trace(tt::LogTest, "Sender 1 num buffers: {}", this->sender_1_num_buffers);
+    log_trace(tt::LogTest, "Sender 1 channel address: {}", this->local_sender_channel_1_buffer_address);
+    log_trace(tt::LogTest, "Receiver num buffers: {}", this->receiver_num_buffers);
+    log_trace(tt::LogTest, "Receiver channel address: {}", this->local_receiver_channel_buffer_address);
+    return std::vector<uint32_t>{
+        is_handshake_master,
+        this->handshake_address,
+        this->channel_buffer_size,
+
+        this->sender_0_num_buffers,
+        this->receiver_num_buffers,
+
+        config.sender_0_channel_base_address,
+        FabricEriscDatamoverConfig::sender_channel_0_worker_connection_info_address,
+        config.sender_1_channel_base_address,
+        FabricEriscDatamoverConfig::sender_channel_1_worker_connection_info_address,
+        config.receiver_channel_base_address,
+        config.receiver_channel_base_address,
+
+        config.sender_0_channel_base_address,
+        config.sender_1_channel_base_address,
+
+        this->termination_signal_ptr};
+}
+
+std::vector<uint32_t> FabricEriscDatamoverBuilder::emit_runtime_args() const {
```

Comment:
wonder whats the semantics of "emit"? 
in qt emit is to emit a signal. this just gives you the output as any other getter /compute function would do

```suggestion
std::vector<uint32_t> FabricEriscDatamoverBuilder::get_runtime_args() const {
```

or 

```suggestion
std::vector<uint32_t> FabricEriscDatamoverBuilder::generate_runtime_args() const {
```

Reply:
sure I don't have a strong opinion on this, I can change it